src/rockstor/storageadmin/views/command.py

```diff
@@ -90,6 +90,16 @@ def _refresh_pool_state():
                     "are no attached devices. Moving on.".format(p.name)
                 )
                 continue
+            # If pool has no missing remove all detached disk pool associations.
+            # Accounts for 'end of run' clean-up in removing a detached disk and for cli
+            # maintenance re pool returned to no missing dev status. Also re-establishes
+            # pool info as source of truth re missing.
+            if not p.has_missing_dev:
+                for disk in p.disk_set.filter(name__startswith="detached-"):
+                    logger.info("Removing detached disk from Pool {}: no missing "
```

Comment:
Quick thought: Could we further detail the disk name (or any meaningful info) on the disk in question here? While testing, two detached disks needed to be removed and that results in the following:
```
[19/Jun/2020 07:59:36] INFO [storageadmin.views.command:100] Removing detached disk from Pool test_pool_03: no missing devices found.
[19/Jun/2020 07:59:36] INFO [storageadmin.views.command:100] Removing detached disk from Pool test_pool_03: no missing devices found.
```

It thus results in duplicates (or tri-, etc...) when that happens. Tiny detail, but if there's some some of identification readily available for the detached disk in question being removed, it could be helpful.

Reply:
Good point except that the name of a detached disk is essentially random. The bit after detached is just a uuid (long historical reason to that) so the name wouldn't help. That random element is also changed on every disk refresh!! And other than that I guess the best thing would be the serial. Needn't hold up the pr merge though as almost all removals will be singular devices anyway given only meaningful removals (of viable pools) in raid 6 will be more than one given it's the only raid level that can handle more than one disk failure currently anyway (bar upstream btrfs raid1c3 and raid1c4).

I say we earmark this as a future logging improvement as we will be back here before long anyway. Nice idea though.